RecoEcal/EgammaClusterProducers/src/EGRefinedSCFixer.cc

```diff
@@ -0,0 +1,491 @@
+  auto fixedEESCs = getHandle(iEvent,fixedSCToken_[1]);
+  auto fixedPFClusters = getHandle(iEvent,fixedPFClustersToken_);
+
+  auto fixedRefinedSCs = std::make_unique<reco::SuperClusterCollection>();
+  // EB clusters are fixed, EE are direct translation from the original
+  auto fixedBCs = std::make_unique<reco::CaloClusterCollection>();
+  // direct translation of the original ES collection - there is nothing "fixed" about this
+  auto fixedESs = std::make_unique<reco::CaloClusterCollection>();
+  auto fixedConvs = std::make_unique<reco::ConversionCollection>();
+
+  std::vector<reco::SuperClusterRef> mappedRefinedSCs;
+  std::vector<reco::SuperClusterRef> mappedSCsEB(fixedEBSCs->size());
+  std::vector<reco::SuperClusterRef> mappedSCsEE(fixedEESCs->size());
+
+  std::vector<edm::Ptr<reco::CaloCluster> > clusterAddedToAnyEBSC;
+  for(auto& refinedSC : *orgRefinedSCs){  
+    if(refinedSC.seed()->seed().subdetId() == EcalBarrel) { 
+      reco::SuperClusterRef parentSC = GainSwitchTools::matchSCBySeedCrys(refinedSC,orgEBSCs);
+      if(parentSC.isNonnull()) {
+	auto clusAdded = getSubClustersMissing(refinedSC,*parentSC);
+	for(auto clusPtr : clusAdded) clusterAddedToAnyEBSC.push_back(clusPtr);
+      }else{ //no parent supercluster, all superclusters in this supercluster are addded
+	for(auto clusPtr : refinedSC.clusters()) clusterAddedToAnyEBSC.push_back(clusPtr);
+      }
+    }
+  }
+    
+  for (unsigned iO(0); iO != orgRefinedSCs->size(); ++iO) {
+    auto& orgRefinedSC(orgRefinedSCs->at(iO));
+    mappedRefinedSCs.emplace_back(orgRefinedSCs, iO);
+
+    // find the PFSuperCluster for this refined SC
+    if (orgRefinedSC.seed()->seed().subdetId() == EcalBarrel) {
+      auto orgEBSC(GainSwitchTools::matchSCBySeedCrys(orgRefinedSC, orgEBSCs));
+
+      // particleFlowEGamma can create superclusters directly out of PFClusters too
+      // -> there is not always a matching EB SC
+      if (orgEBSC.isNonnull()) {
+	//changing the matching to be in  3x3 now (had issues with suprious matches)
+        auto fixedEBSC(GainSwitchTools::matchSCBySeedCrys(*orgEBSC, fixedEBSCs, 1, 1));
+       
+        // here we may genuinely miss a mapping, if the seed position moves too much by re-reconstruction
+	// Sam: its very unlikely, if not impossible. To be replaced the gain switched crystal must be within +/-1 crystal a hybrid supercluster seed crystal because of ecalSelectedDigis. You could only get a shift larger than 1 if you had two supercluster seed crystals very close together and even then I'm not sure its possible. 
+        if (fixedEBSC.isNonnull()) {
+          mappedSCsEB[fixedEBSC.key()] = orgEBSC;
+
+          auto fixedRefinedSC(makeFixedRefinedBarrelSC(orgRefinedSC, *orgEBSC, *fixedEBSC, fixedPFClusters,clusterAddedToAnyEBSC));
+          fixedRefinedSCs->push_back(fixedRefinedSC);
+
+	  //	  std::cout<<" org refined "<<SCDeepPrinter(orgRefinedSC)<<std::endl;
+	  //std::cout<<" new refined "<<SCDeepPrinter(fixedRefinedSC)<<std::endl;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant